librz/analysis/var.c

```diff
@@ -203,6 +203,24 @@ RZ_API void rz_analysis_function_delete_vars_by_storage_type(RzAnalysisFunction
 	}
 }
 
+/**
+ * Delete all variables from \p fcn that are either arguments or locals, depending on \p args
+ * \param args if true, all args will be deleted, otherwise all non-arg vars will be deleted
+ */
+RZ_API void rz_analysis_function_delete_vars_by_arg_property(RzAnalysisFunction *fcn, bool args) {
```

Comment:
`args` parameter is not used, this function currently always deletes args, while the parameter says it will flip between args or not. This function is only used in one place where this is in fact the case.
There is the option of just removing the parameter, in which case the function is better called `rz_analysis_function_delete_all_args` or `rz_analysis_function_delete_arg_vars`.
Alternatively, if select specific filters are desired, it may be clearer to provide an enum or flags, allowing more filter options in the future. In this case a fitting name would be `rz_analysis_function_delete_vars` (as opposed to the "all_vars" version, though merging these functions to some degree would make sense), or `rz_analysis_function_delete_vars_matching`.

Reply:
Good catch, I will make it `rz_analysis_function_delete_arg_vars()`. The fact that the arg slipped through already shows this would have been more complexity than needed.

@wargio Recreating the entire function would not work for this use-case because you want to keep all (e.g. manually renamed and typed) local variables, control flow analysis, etc.